server/src/database/archive.rs

```diff
@@ -90,3 +91,88 @@ pub(crate) fn fetch_transactions(
         .get_results(connection)?;
     Ok(results)
 }
+
+use std::collections::HashMap;
+
+use graphql_client::{reqwest::post_graphql_blocking as post_graphql, GraphQLQuery, Response};
+use reqwest::blocking::Client;
+
+use transaction_query::TransactionQueryTransactions;
+
+type DateTime = String;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "src/database/transaction_schema.graphql",
+    query_path = "src/database/transaction_query.graphql",
+    response_derives = "Debug"
+)]
+pub struct TransactionQuery;
+
+#[allow(clippy::unwrap_used, clippy::upper_case_acronyms)]
+pub(crate) fn fetch_transactions_graphql(
+    start_time_millis: i64,
+    end_time_millis: i64,
+    mip_key: &str,
+    base_memo: &str, // Pass the base memo as an argument
+) -> Result<Vec<FetchTransactionResult>> {
+    let start_unix_timestamp = start_time_millis / 1000; // Convert milliseconds to seconds
+    let end_unix_timestamp = end_time_millis / 1000;     // Convert milliseconds to seconds
+
+    let start_datetime = OffsetDateTime::from_unix_timestamp(start_unix_timestamp);
+    let end_datetime = OffsetDateTime::from_unix_timestamp(end_unix_timestamp);
+
+    let start_utc_datetime = start_datetime.expect("not a valid time value").to_offset(UtcOffset::UTC);
+    let end_utc_datetime = end_datetime.expect("not a valid time value").to_offset(UtcOffset::UTC);
+
+    let lower_case_memo = base_memo.to_string();
+    let upper_case_memo = base_memo.to_uppercase();
+    let no_lower_case_memo = format!("no {:?}", upper_case_memo);
+    let no_upper_case_memo = format!("NO {:?}", upper_case_memo);
+
+    let variables = transaction_query::Variables {
+        date_time_gte: Some(start_utc_datetime.to_string()),
+        date_time_lte: Some(end_utc_datetime.to_string()),
+        memo1: Some(lower_case_memo),
+        memo2: Some(upper_case_memo),
+        memo3: Some(no_lower_case_memo),
+        memo4: Some(no_upper_case_memo),
+    };
```

Comment:
The memo values need to be b58 memo encoded. I created those functions in the graphql-mina-explorer project

Reply:
addressed in recent commit below https://github.com/Granola-Team/mina-on-chain-voting/pull/237/commits/9ad42b36e00e9e37033dea9381e111c065027060